File: DukeMapGenLib/src/java/trn/duke/experiments/SpritePrefab.java
package trn.duke.experiments;

import trn.Sprite;

/**
 * TODO get rid of this
 */
public class SpritePrefab {

	Integer texture;
	
	Short pallette;
	
	public SpritePrefab(int texture){
		this.texture = Integer.valueOf(texture);
	}
	
	public SpritePrefab setPal(short pallette){
		this.pallette = Short.valueOf(pallette);
		return this;
	}
	
	public void writeTo(Sprite s){
		if(texture != null){
			s.setTexture(texture);
		}
		
		if(pallette != null){
			s.setPal(pallette);
		}
	}
}

File: DukeMapGenLib/src/scala/trn/prefab/SpritePrefab.scala
package trn.prefab

import trn.Sprite


/**
  * TODO rename to SpriteBrush
  */
trait SpritePrefab {
  def tex: Int
  def pal: Int
  def lotag: Int = 0
  def hitag: Int = 0

  final def writeToSprite(sprite: Sprite): Unit = {
    sprite.setTexture(tex)
    sprite.setPal(pal)
    sprite.setLotag(lotag)
    sprite.setHiTag(hitag)
  }
}
